components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -17761,6 +17762,736 @@ public void deleteTenantTheme(int tenantId) throws APIManagementException {
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, organization);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    return resultSet.getString(1);
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Add artifact to AM_ARTIFACT Table.
+     *
+     * @param connection   DB connection.
+     * @param themeContent Theme content.
+     * @param artifactType Artifact's type.
+     * @return String of artifact's ID.
+     */
+    private String addArtifact(Connection connection, InputStream themeContent, String artifactType) throws SQLException {
+        String id = UUID.randomUUID().toString();
+        String query = SQLConstants.DevPortalContentConstants.ADD_ARTIFACT;
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, id);
+            preparedStatement.setBinaryStream(2, themeContent);
+            preparedStatement.setString(3, artifactType);
+            preparedStatement.executeUpdate();
+            return id;
+        }
+    }
+
+    /**
+     * Get artifact content from AM_ARTIFACT Table.
+     *
+     * @param connection DB connection.
+     * @param artifactId Artifact's ID.
+     * @return Artifact content.
+     */
+    private InputStream getArtifactContent(Connection connection, String artifactId) throws SQLException {
+        String query = SQLConstants.DevPortalContentConstants.GET_ARTIFACT;
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, artifactId);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    return resultSet.getBinaryStream(1);
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Remove artifact from AM_ARTIFACT Table.
+     *
+     * @param connection DB connection.
+     * @param artifactId Artifact's ID.
+     */
+    private void removeArtifact(Connection connection, String artifactId) throws SQLException {
+        String query = SQLConstants.DevPortalContentConstants.DELETE_ARTIFACT;
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, artifactId);
+            preparedStatement.executeUpdate();
+        }
+    }
+
+    /**
+     * Insert a fresh Org with drafted artifact's ID to AM_DEVPORTAL_ORG_CONTENT.
+     *
+     * @param connection   DB connection.
+     * @param organization Organization name.
+     * @param artifactUUID Artifact's ID.
+     */
+    private void insertNewOrgWithDraftedArtifact(Connection connection, String organization, String artifactUUID) throws SQLException {
```

Comment:
Do we have to specifically mention "New"? As it is an org insert, I don't think we need to append "New" to the method call?

Reply:
Changed through https://github.com/wso2/carbon-apimgt/pull/12874/commits/439ab2cdf303bea83a11ef850970dd032e5c9f00